com.unity.visualeffectgraph/Editor/Models/VFXGraph.cs

```diff
@@ -105,7 +105,20 @@ static void OnCompileResource(VisualEffectResource resource)
                     }
                     else
                     {
-                        resource.GetOrCreateGraph().CompileForImport();
+                        //Workaround, use backup system to prevent any modification of the graph during compilation
+                        //The responsible of this unexpected change is PrepareSubgraphs => RecurseSubgraphRecreateCopy => ResyncSlots
+                        //It will let the VFXGraph in a really bad state after compilation.
+                        graph = resource.GetOrCreateGraph();
+                        var dependencies = new HashSet<ScriptableObject>();
+                        dependencies.Add(graph);
+                        graph.CollectDependencies(dependencies);
+                        var backup = VFXMemorySerializer.StoreObjectsToByteArray(dependencies.ToArray(), CompressionLevel.None);
+
+                        graph.CompileForImport();
```

Comment:
But still this means that CompileForImport can still trigger a SetDirty during compilation, which can potentially trigger, another useless import, or worse an infinite compile?

Reply:
The dirty flag (which isn't serialized) about compilation is cleared at the end of compilation.
```
m_ExpressionGraphDirty = false;
m_ExpressionValuesDirty = false;
```
The issue is due to preprocess we are doing *before* the actual compilation.
The backup doesn't restore these dirty flags.
Then, the [SetDirty](https://docs.unity3d.com/ScriptReference/EditorUtility.SetDirty.html) which can occurs (maybe ?) isn't detected, otherwise, I would have a infinite compilation in the editor test I introduced.